src/serializer.rs

```diff
@@ -275,7 +290,32 @@ macro_rules! impl_tocss_for_int {
     ($T: ty) => {
         impl<'a> ToCss for $T {
             fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-                write!(dest, "{}", *self)
+                struct AssumeUtf8<W: fmt::Write>(W);
+
+                impl<W: fmt::Write> io::Write for AssumeUtf8<W> {
+                    #[inline]
+                    fn write_all(&mut self, buf: &[u8]) -> io::Result<()> {
+                        // Safety: itoa only emits ASCII
```

Comment:
Can we assert it?

```rust
debug_assert!(buf.iter().all(|b| b.is_ascii()));
```

Reply:
Done. (`AsciiExt` is also implemented on `[u8]`, no need to iterate ourselves.)